### PR TITLE
feat:complete contributors dialog and table of contents

### PR DESCRIPTION
**Pull request by JeffreyMa597**
What problem does this PR solve?
------
As title.

Ref: https://github.com/Magickbase/nervos-official-website/issues/62

Check List
------

### Test
e2e Test
### Task
none

### CODE REVIEW

**Comment by Keith-CY** [comment]
CI failed at

**Comment by WhiteMinds** [comment]
Currently not following page theme changes

image


image

File: src/components/BaseSeparatePage/TableOfContents/useHeadingsData.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * Dynamically generates the table of contents list
+ */
+
+export type NestedHeadingtype = {
+  id: string
+  title: string
+  items?: NestedHeadingtype[]
+}
+
+const useHeadingsData = () => {
```
Comment (WhiteMinds) [inline src/components/BaseSeparatePage/TableOfContents/useHeadingsData.ts:13]: This implementation has too much implicit coupling and is only globally singleton, requiring refactoring

File: src/components/BaseSeparatePage/Info/index.tsx [hunk 2]
```diff
@@ -13,13 +13,14 @@ export type InfoType = {
...
+  onContributorsButtonClick?: () => void
 }
 
-export const Info: FC<InfoType> = ({ info, editor, editLink, className }) => {
+export const Info: FC<InfoType> = ({ info, editor, editLink, className, onContributorsButtonClick }) => {
   const time = new Date().toDateString()
 
   const ContributorsButton = () => (
-    <button onClick={() => alert('SEE CONTRIBUTORS')} className={styles.contributorsButton}>
+    <button onClick={onContributorsButtonClick} className={styles.contributorsButton}>
```
Comment (WhiteMinds) [inline src/components/BaseSeparatePage/Info/index.tsx:23]: Why not just put the ContributorsDialog inside this component?

File: src/components/BaseSeparatePage/FunctionsItem/index.tsx [hunk 3]
```diff
@@ -6,15 +6,20 @@ import styles from './index.module.scss'
...
 }
 
 const TagItem = ({ tag }: { tag: string }) => <div className={clsx(styles.tagItem)}>{tag}</div>
 
-export const FunctionsItem: FC<FunctionsItemType> = ({ title, tags, content, isProgressBar, className }) => (
-  <div data-is-progress-bar={isProgressBar} className={clsx(styles.functionsItem, className)}>
+export const FunctionsItem: FC<FunctionsItemType> = ({ title, tags, content, isProgressBar, idx, className }) => (
+  <div
+    data-is-progress-bar={isProgressBar}
+    id={typeof title === 'string' ? title : `functions-${idx!}`}
```
Comment (WhiteMinds) [inline src/components/BaseSeparatePage/FunctionsItem/index.tsx:20]: What is the purpose of this id?

**Review by WhiteMinds — CHANGES_REQUESTED** [review]

File: src/components/BaseSeparatePage/FunctionsItem/index.tsx (hunk 3 above)
Comment (JeffreyMa597) [reply src/components/BaseSeparatePage/FunctionsItem/index.tsx:20]: This is for scrollIntoView the element.

File: src/components/BaseSeparatePage/Info/index.tsx (hunk 2 above)
Comment (JeffreyMa597) [reply src/components/BaseSeparatePage/Info/index.tsx:23]: emmm.... both of them are LGTM. But I thought it would be clearer to make it a child component.

**Pull request merged**